pkg/controller/controller.go

```diff
@@ -595,9 +595,24 @@ func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.Persist
 		return nil, controller.ProvisioningFinished, err
 	}
 
+	pvcAnnotationsForSCParam := map[string]string{}
+	for k, v := range claim.GetAnnotations() {
+		pvcAnnotationsForSCParam["pvc.annotations['"+k+"']"] = v
+	}
+
 	fsTypesFound := 0
 	fsType := ""
 	for k, v := range sc.Parameters {
+		// Replace SC parameters with PVC annoations
+		if strings.Contains(v, "pvc.annotations") {
+			resolvedName, err := resolveTemplate(v, pvcAnnotationsForSCParam)
+
+			if err != nil {
+				return nil, controller.ProvisioningFinished, fmt.Errorf("error resolving value %q: %v", v, err)
```

Comment:
are you sure you want to return error here just because resolving value failure, I think it's better logging here.

Reply:
If a parameter follows "${pvc.annoations" pattern but it can't find any match in pvc.annoations, it should be a human mistake, either bad annotation name in SC parameter or missing annotation in pvc.annoations.

Given that, I think it's better to error it out.